### ISSUE TITLE
WebDAV plugin: No HTTP-timeout leads to indefinite blocking when connection is gone

### ISSUE THREAD

**Issue by binwiederhier** [syncany/syncany#305]
When I played around with the resume-functionality (see #141), I noticed that the WebDAV plugin apparently doesn't have a HTTP timeout. Taking away the WiFi when `sy up -d` is running (and files are being uploaded) leads to no exceptions, no termination of the program -- all threads just block and wait for the connection to come back. When it does, it actually continues. 

However, the expected behavior is that the connection should break, so that resume can kick in the next time it starts. 

Long story short: The WebDAV/HTTP connection needs a timeout.
